src/util/transport.rs

```diff
@@ -41,25 +55,33 @@ impl<T: Debug> SendCh<T> {
         SendCh { ch: ch }
     }
 
-    pub fn send(&self, mut t: T) -> Result<(), Error> {
-        let mut try_cnt = 0;
-        while let Err(e) = self.ch.send(t) {
-            t = match e {
-                NotifyError::Full(m) => m,
-                _ => return Err(box_err!("{:?}", e)),
+    pub fn send(&self, t: T) -> Result<(), Error> {
+        self.send_with_try_times(t, 1)
+    }
+
+    /// Try send t with default try times.
+    pub fn try_send(&self, t: T) -> Result<(), Error> {
+        self.send_with_try_times(t, MAX_SEND_RETRY_CNT)
```

Comment:
I prefer send using retry count but try_send using only 1, like channel send and try_send.